sweepai/api.py

```diff
@@ -234,6 +193,82 @@ def call_on_merge(*args, **kwargs):
     thread.start()
 
 
+function_mapping = {
+    "on_merge": on_merge,
+    "on_merge_conflict": on_merge_conflict,
+    "stack_pr": stack_pr,
+    "make_pr": make_pr,
+    "handle_button_click": handle_button_click,
+}
+
+
+@dataclass
+class Task:
+    handler_name: str
+    args: tuple = ()
+    kwargs: dict = field(default_factory=dict)
+
+
+user_queues: dict[str, "Queue[Task]"] = SafeDictionary()
+active_workers: dict[str, 0] = SafeDictionary()
+
+
+def display_queue():
+    result = "Queue:\n"
+    for key, q in list(user_queues.items()):
+        items = ", ".join([task.handler_name for task in q.queue])
+        result += f"{key}: {items}\n"
+    result += str(active_workers._dict)
+    return result
+
+
+def worker(username):
+    while True:
+        user_queue = user_queues.get(username)
+        if user_queue is None:
+            break
+
+        try:
+            task = user_queue.get(timeout=2)
+        except queue.Empty:
+            if user_queues[username].empty():
+                active_workers[username] = active_workers.get(username, 0) - 1
+                break
+            continue
+
+        try:
+            function_mapping[task.handler_name](*task.args, **task.kwargs)
+        except Exception as e:
+            logger.error(f"Exception {e}")
+        user_queue.task_done()
```

Comment:
**Enhance worker function with improved error handling and logging for better debugging and robustness.** 
 
 Enhance the worker function with better error handling and logging. This will improve debugging and make the system more robust. 
```suggestion
def worker(username):
    while True:
        user_queue = user_queues.get(username)
        if user_queue is None:
            logger.info(f"Worker for {username} shutting down: queue not found")
            break

        try:
            task = user_queue.get(timeout=2)
        except queue.Empty:
            if user_queues[username].empty():
                active_workers[username] = active_workers.get(username, 0) - 1
                logger.info(f"Worker for {username} shutting down: no more tasks")
                break
            continue

        logger.info(f"Processing task for {username}: {task.handler_name}")
        try:
            function_mapping[task.handler_name](*task.args, **task.kwargs)
        except KeyError:
            logger.error(f"Unknown task handler: {task.handler_name}")
        except Exception as e:
            logger.exception(f"Error processing task {task.handler_name} for {username}: {e}")
        else:
            logger.info(f"Task {task.handler_name} completed for {username}")
        finally:
            user_queue.task_done()
```